great_expectations/datasource/fluent/spark_dbfs_datasource.py

```diff
@@ -0,0 +1,82 @@
+from __future__ import annotations
+
+import logging
+import re
+from typing import TYPE_CHECKING, Optional, Union
+
+from typing_extensions import Literal
+
+from great_expectations.core._docs_decorators import public_api
+from great_expectations.core.util import DBFSPath
+from great_expectations.datasource.fluent import SparkFilesystemDatasource
+from great_expectations.datasource.fluent.data_asset.data_connector import (
+    DBFSDataConnector,
+)
+from great_expectations.datasource.fluent.spark_file_path_datasource import (
+    CSVAsset,
+)
+
+if TYPE_CHECKING:
+    from great_expectations.datasource.fluent.interfaces import (
+        Sorter,
+        SortersDefinition,
+    )
+
+logger = logging.getLogger(__name__)
+
+
+@public_api
+class SparkDBFSDatasource(SparkFilesystemDatasource):
+    """Spark based Datasource for DataBricks File System (DBFS) based data assets."""
+
+    # instance attributes
+    type: Literal["spark_dbfs"] = "spark_dbfs"
+
+    @public_api
+    def add_csv_asset(
+        self,
+        name: str,
+        batching_regex: Optional[Union[re.Pattern, str]] = None,
+        glob_directive: str = "**/*",
```

Comment:
Can `glob_directive` be part of the `Datasource` parameters?

Right now it's not a field on the `DataAsset` or `Datasource` and only exists on this manually defined `add_csv_asset()` (etc.) methods.

We are trying to get aways from having to manually define these `add_<ASSET_TYPE>_asset()` and the fact that `glob_directory` isn't part of the `asset` or `datasource` presents a problem when trying to generalize this.

Reply:
@Kilo59 I am happy to discuss, but in general, `glob` has some aspects of regex, but we are using regex for partitioning.  Hence, in order to move `glob_directive` to `Datasource` universally, we will have to clarify how we explain it as well as modify a large number of tests.  Let us discuss it in an open idea-generating forum and not be haste with a decision, until also, perhaps conducting some preliminary experiments in order to fully measure the impact.  Thank you.